src/lxp/inverter.rs

```diff
@@ -130,17 +130,24 @@ pub struct Inverter {
     config: ConfigWrapper,
     host: String,
     channels: Channels,
+    receiver: std::cell::Cell<Option<Receiver>>,
 }
 
 impl Inverter {
     pub fn new(config: ConfigWrapper, inverter: &config::Inverter, channels: Channels) -> Self {
         // remember which inverter this instance is for
         let host = inverter.host().to_string();
 
+        // Create the receiver at new() time, rather than start(), so things
+        // wanting to send to the inverter queue messages, rather than failing
+        // with an error about closed channels
+        let receiver = std::cell::Cell::new(Some(channels.to_inverter.subscribe()));
```

Comment:
FWIW I was quite surprised to discover there is only one channel for all inverters. In my head, it was going to be a channel per inverter.

I don't know if it makes a practical difference for this PR, but it was giving me all kinds of grief and edge cases when fiddling with `receiver_count`.

Reply:
I've wondered about changing this before as I agree its surprising but there's never been a compelling case to do so. So it would have been change for the sake of it.

I wouldn't be opposed to changing that if it doesn't add too much extra complication.

Reply:
I'd rather leave it out of this PR if that's alright with you :sweat_smile:. If we were going the `receiver_count` route I'd definitely have been doing it, but this way it's neither here nor there, I think.